apps/src/dance/DanceAiClient.ts

```diff
@@ -0,0 +1,101 @@
+import UntypedCachedPalettes from '@cdo/static/dance/ai/model/cached-spacy-palette-map.json';
+import UntypedCachedBackgrounds from '@cdo/static/dance/ai/model/cached-spacy-background-map.json';
+import UntypedCachedForegrounds from '@cdo/static/dance/ai/model/cached-spacy-foreground-map.json';
+
+// Custom types for vector mapping JSONs to play nice with Typescript
+type EmojiAssociation = {
+  association: number[];
+};
+
+type CachedMapping = {
+  emojiAssociations: {[key: string]: EmojiAssociation};
+  output: string[];
+};
+
+const CachedPalettes: CachedMapping = UntypedCachedPalettes;
+const CachedBackgrounds: CachedMapping = UntypedCachedBackgrounds;
+const CachedForegrounds: CachedMapping = UntypedCachedForegrounds;
+
+// Output: JSON containing {backgroundColor:string, backgroundEffect:string, foregroundEffect:string}
+// ToDo: Break down into helper functions for each step rather than repeat code for each field
+export default function cachedAi(input: string) {
```

Comment:
Separately, I recommend changing this so it is not a "default" export.

Reasoning: Given the filename DanceAiClient, I would expect this to be an API client, which it appears to be (but with the only action right now to be shortcutting the API call in favor of a cached version). With that, I would expect any default export (if there is one) to be a class. What would also seem normal would be to import the module and call available functions. For example, DanceAiClient.cachedAi(input). Importing this directly as DanceAiClient would make me expect that I get the whole client rather than a single function that returns the 3 specific effects.

Caveat: I'm not as familiar with TS as I am with vanilla JS, so if you know/find something that suggests a different convention, I'm happy to be proven wrong. I did find the [TS guide](https://www.typescriptlang.org/docs/handbook/modules.html), which seems to suggest (although not declare) that the above approach would be correct.

Reply:
Let's review this as a next step but I generally agree that it seems strange to import a whole class just for one function, even if that class has a fair number of dependencies; I'm also unfamiliar with typescript protocols in that regard

Reply:
Consolidating this with the trello ticket task above to generally just migrate this into the utils file: https://trello.com/c/KvfZFqlX